packages/markdown-docx/src/OOXMLTransformer/index.js

```diff
@@ -58,200 +87,214 @@ class OoxmlTransformer {
+        let obj = {};
+        if (nodeInformation.nodeType === 'softbreak') {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Softbreak`,
+            };
+        } else if (nodeInformation.nodeType === 'variable') {
+            if (nodeInformation.elementType && nodeInformation.name && nodeInformation.value) {
+                obj = {
+                    $class: `${NS_PREFIX_CiceroMarkModel}Variable`,
+                    value: nodeInformation.value,
+                    elementType: nodeInformation.elementType,
+                    name: nodeInformation.name,
+                };
+            } else {
+                return {};
             }
+        } else {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Text`,
+                text: nodeInformation.value,
+            };
         }
-        return null;
+        for (let i = nodeInformation.properties.length - 1; i >= 0; i--) {
+            obj = {
+                $class: nodeInformation.properties[i],
+                nodes: [obj],
+            };
+        }
+        return obj;
     }
 
     /**
-     * Checks if the element is a line break or not
+     * Generates all nodes present in a block element( paragraph, heading ).
      *
-     * @param {object} element the element to be checked
-     * @returns {boolean} whether the element is a line break
+     * @param {object} rootNode Block node like paragraph, heading, etc.
      */
-    isLineBreak(element) {
-        return element.name === 'w:p' && element.elements === undefined;
-    }
+    generateNodes(rootNode) {
+        if (this.JSONXML.length > 0) {
+            let constructedNode;
+            constructedNode = this.constructCiceroMarkNode(this.JSONXML[0]);
+            rootNode.nodes = [...rootNode.nodes, constructedNode];
 
-    /**
-     * Returns if the element is a heading or not and if it is, an attribute
-     * level is also passed to determine the style of the heading.
-     *
-     * @param {object} headingElement the element to be checked
-     * @returns {object} object of `isHeading` and its level
-     */
-    getHeading(headingElement) {
-        if (headingElement.attributes !== undefined) {
-            const headingLevel = headingElement.attributes['w:val'];
-            if (headingElement.name === 'w:pStyle' && headingLevel.includes('Heading')) {
-                return {
-                    isHeading: true,
-                    level: headingLevel[headingLevel.length - 1],
+            let rootNodesLength = 1;
+            for (let nodePropertyIndex = 1; nodePropertyIndex < this.JSONXML.length; nodePropertyIndex++) {
+                let propertiesPrevious = this.JSONXML[nodePropertyIndex - 1].properties;
+                let propertiesCurrent = this.JSONXML[nodePropertyIndex].properties;
+
+                let commonPropertiesLength = 0;
+                for (let j = 0; j < Math.min(propertiesPrevious.length, propertiesCurrent.length); j++) {
```

Comment:
Could you be cleared what `j` means? Please spend some time naming your variables.